ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -258,6 +262,44 @@ RaftGroup getGroup() {
     return RaftGroup.valueOf(getMemberId().getGroupId(), getRaftConf().getPeers());
   }
 
+  /**
+   * This removes the group from the server.
+   * If the flag is set to false, the directory is renamed to a random uuid
+   * along with {@value GROUP_REMOVE_DETAILS_FILE} which holds the information
+   * about its original group.
+   * If the flag is true, the group is permanently deleted.
+   * @param deleteDirectory
+   */
+  public void groupRemove(boolean deleteDirectory) {
+    final RaftStorageDirectory dir = state.getStorage().getStorageDir();
+
+    /* Do not delete the directory here irrespective of the flag.
+    Shutdown is triggered inorder to avoid any locked files. */
+    shutdown(false);
+
+    /* The group is renamed to a random uuid and a properties file is stored. */
+    if (!deleteDirectory) {
```

Comment:
I think we can create a directory to contain all removed groups, and the removed group in the directory should keep their name.

Reply:
Fair enough. This way there is no need to create extra file. I will update the patch.